### PR TITLE
PULSEBS-55 add endpoint to set lecture as online

### PR DESCRIPTION
**Pull request by Francexi**

### CODE REVIEW

File: server/API/REST.php [hunk 1]
```diff
@@ -590,6 +589,75 @@ function cancel_booking($vars) {
...
+			$nextHour->modify('+30 minutes'); // Check for timezones discrepancies
+			$stmt = $pdo->prepare('SELECT *
+								   FROM lectures L, courses C
+								   WHERE L.course_id = C.ID
+									   AND L.ID = :lectureId
+									   AND C.teacher_id = :userId
+									   AND L.start_ts > :nextHour');
+			$stmt->bindValue(':lectureId', $lectureId, PDO::PARAM_INT);
+			$stmt->bindValue(':userId', $userId, PDO::PARAM_INT);
+			$stmt->bindValue(':nextHour', $nextHour->getTimestamp(), PDO::PARAM_INT);
```
Comment (LeaX-XIV) [inline server/API/REST.php:628]: Wrong variable name, should be next30mins, but its irrelevant to code execution

File: server/API/REST.php [hunk 2]
```diff
@@ -590,6 +589,75 @@ function cancel_booking($vars) {
...
+								   WHERE ID = :lectureId');
+			}
+			else{
+				$stmt = $pdo->prepare('UPDATE lectures
+								   SET settings = settings & :online
+								   WHERE ID = :lectureId');
+
+			}
+			$stmt->bindValue(':online', $status, PDO::PARAM_INT);
+			$stmt->bindValue(':lecture', $lectureId, PDO::PARAM_INT);
```
Comment (LeaX-XIV) [inline server/API/REST.php]:
```suggestion
			$stmt->bindValue(':lectureId', $lectureId, PDO::PARAM_INT);
```

File: server/API/REST.php [hunk 3]
```diff
@@ -590,6 +589,75 @@ function cancel_booking($vars) {
...
+				$stmt = $pdo->prepare('UPDATE lectures
+								   SET settings = settings | :online
+								   WHERE ID = :lectureId');
+			}
+			else{
+				$stmt = $pdo->prepare('UPDATE lectures
+								   SET settings = settings & :online
+								   WHERE ID = :lectureId');
+
+			}
```
Comment (LeaX-XIV) [inline server/API/REST.php:647]: Is this the same as `UPDATE lectures SET settings = 0 WHERE ID = :lectureId`? I have trouble understanding it

File: server/API/REST.php (hunk 2 above)
Comment (LeaX-XIV) [inline server/API/REST.php]:
```suggestion
			$stmt->bindValue(':lectureId', $lectureId, PDO::PARAM_INT);
```

File: server/API/REST.php (hunk 3 above)
Comment (Francexi) [reply server/API/REST.php:647]: no. If you like that, you clear all the settings bit. This way you only write the correct bit

**Pull request merged**